src/java.net.http/share/classes/jdk/internal/net/http/common/HttpBodySubscriberWrapper.java

```diff
@@ -75,6 +73,15 @@ public HttpBodySubscriberWrapper(BodySubscriber<T> userSubscriber) {
         this.userSubscriber = userSubscriber;
     }
 
+    /**
+     * A callback to be invoked before termination, whether due to the
+     * completion or failure of the subscriber, or cancellation of the
+     * subscription.
```

Comment:
I'd suggest to also say that when a subscription is cancelled, onTermination() is called before onCancel().

Reply:
Improved docs in 7c989980e19.